From: Jack Steiner <steiner@sgi.com>
Subject: x86, UV: Fix interrupt targeting for GRU on Westmere
Patch-mainline: Not yet
References: bnc#662340

Add UV function to determine the blade-local processor
socket number from an apicid.

Signed-off-by: Jack Steiner <steiner@sgi.com>
Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
---
 arch/x86/include/asm/uv/uv_hub.h |    8 ++++++++
 1 file changed, 8 insertions(+)

Index: linux-2.6.32-SLE11-SP1/arch/x86/include/asm/uv/uv_hub.h
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/arch/x86/include/asm/uv/uv_hub.h
+++ linux-2.6.32-SLE11-SP1/arch/x86/include/asm/uv/uv_hub.h
@@ -311,6 +311,14 @@ static inline int uv_apicid_to_pnode(int
 }
 
 /*
+ * Convert an apicid to the socket number on the blade
+ */
+static inline int uv_apicid_to_socket(int apicid)
+{
+	return (apicid >> (uv_hub_info_extra->apic_pnode_shift - 1)) & 1;
+}
+
+/*
  * Access global MMRs using the low memory MMR32 space. This region supports
  * faster MMR access but not all MMRs are accessible in this space.
  */
